pgjdbc/src/main/java/org/postgresql/jdbc/PgResultSetMetaData.java

```diff
@@ -218,49 +218,38 @@ private void fetchFieldMetaData() throws SQLException {
             + "JOIN pg_catalog.pg_attribute a ON (c.oid = a.attrelid) "
             + "JOIN pg_catalog.pg_type t ON (a.atttypid = t.oid) "
             + "LEFT JOIN pg_catalog.pg_attrdef d ON (d.adrelid = a.attrelid AND d.adnum = a.attnum) "
-            + "JOIN (");
+            + "JOIN (VALUES ");
 
-    // 7.4 servers don't support row IN operations (a,b) IN ((c,d),(e,f))
-    // so we've got to fake that with a JOIN here.
-    //
-    boolean hasSourceInfo = false;
+    boolean needsComma = false;
     Set<String> oidSet = new HashSet<>();
     for (Field field : fields) {
       if (field.getMetadata() != null) {
         continue;
       }
-
-      if (hasSourceInfo) {
-        sql.append(" UNION ALL ");
+      if (needsComma) {
+        sql.append(", ");
       }
+      sql.append("( ");
 
-      sql.append("SELECT ");
       sql.append(field.getTableOid());
-      if (!hasSourceInfo) {
-        sql.append(" AS oid ");
-      }
       sql.append(", ");
       sql.append(field.getPositionInTable());
-      if (!hasSourceInfo) {
-        sql.append(" AS attnum");
-      }
-
-      if (!hasSourceInfo) {
-        hasSourceInfo = true;
-      }
+      sql.append(")");
+      needsComma = true;
       oidSet.add(String.valueOf(field.getTableOid()));
     }
-    sql.append(") vals ON (c.oid = vals.oid AND a.attnum = vals.attnum) ");
+    sql.append(") as vals (oid,attnum) ON (c.oid = vals.oid AND a.attnum = vals.attnum) ");
 
     if (!oidSet.isEmpty()) {
       sql.append("where c.oid in (").append(String.join(",", oidSet)).append(")");
     }
 
+    /*
     if (!hasSourceInfo) {
       fieldInfoFetched = true;
       return;
     }
-
+*/
```

Comment:
Why removing this?

Previously, `fieldInfoFetched` was set to `true` in case metadata for all the fields was known.
I guess it was there to prevent duplicate queries from `fetchFieldMetaData()`